src/hooks/useOrds.ts

```diff
@@ -598,6 +611,157 @@ export const useOrds = () => {
     }
   };
 
+  const calculateFee = (numPaymentUtxos: number, purchaseTx: Transaction) => {
+    const byteSize = Math.ceil(P2PKH_INPUT_SIZE * numPaymentUtxos + purchaseTx.to_bytes().byteLength);
+    return Math.ceil(byteSize * FEE_PER_BYTE);
+  };
+
+  const purchaseGlobalOrderbookListing = async (purchaseOrdinal: PurchaseOrdinal) => {
+    try {
+      const { marketplaceAddress, marketplaceRate, outpoint, password } = purchaseOrdinal;
+      if (!bsvWasmInitialized) throw Error('bsv-wasm not initialized!');
+      setIsProcessing(true);
+
+      const isAuthenticated = await verifyPassword(password);
+      if (!isAuthenticated) {
+        return { error: 'invalid-password' };
+      }
+      const keys = await retrieveKeys(password);
+
+      if (!keys.walletWif || !keys.ordWif) return { error: 'no-keys' };
+      const fundingAndChangeAddress = bsvAddress;
+
+      const fundingUtxos = await getUtxos(fundingAndChangeAddress);
+
+      if (!fundingUtxos.length) {
+        throw new Error('Could not retrieve funding UTXOs');
+      }
+
+      const payPk = PrivateKey.from_wif(keys.walletWif);
+      const res = await getUtxoByOutpoint(outpoint);
+      const { vout, txid } = res;
+      const price = Number(res.data?.list?.price);
+      const payout = res.data?.list?.payout;
+
+      const { script } = await getMarketData(outpoint);
+
+      if (!txid || !price || !payout || !script) throw Error('Missing information!');
+
+      const purchaseTx = new Transaction(1, 0);
+
+      const neededAmount = fundingUtxos.reduce((a, r) => a + r.satoshis, 0) + P2PKH_INPUT_SIZE * fundingUtxos.length;
```

Comment:
This seems wrong. You're adding satoshis + size?

Reply:
@shruggr  I followed pattern from 1sat website. I felt like it was wrong as well.